### ISSUE TITLE
refactor Build-System to use recent MSBuild /NuGet

### ISSUE THREAD

**Issue by lytico** [dotdevelop/dotdevelop#53]
### Problem

MonoDevelop's Build System is not future save.
it contains a lot of workarounds of old .net-versions, and code from that time when MSBuild was closed source
at least it's broken already if .net5 is used.

affected projects:
https://github.com/dotdevelop/dotdevelop/tree/master/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild
https://github.com/dotdevelop/dotdevelop/tree/master/main/src/core/MonoDevelop.Projects.Formats.MSBuild

and for NuGet:
https://github.com/dotdevelop/dotdevelop/tree/master/main/src/addins/MonoDevelop.PackageManagement

### possible solution

there is a lot of similar code around, actively supported
but most of it not public accesible (internal, private etc.)

clone this repos, branch it & make usable code accessible  (eg. internal visible to, wrapper project) or adopt it

MSBuild:
https://github.com/dotdevelop/msbuild (partly cloned in https://github.com/dotdevelop/dotdevelop/tree/master/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild)

NuGet:
https://github.com/NuGet/NuGet.Client
https://github.com/NuGet/NuGet.Client/tree/dev/src/NuGet.Clients/NuGet.PackageManagement.UI (partly cloned in https://github.com/dotdevelop/dotdevelop/tree/master/main/src/addins/MonoDevelop.PackageManagement)

etc. see NuGet.VisualStudio ff

### deliver own versions
there should be local, proved working fallback versions packaged with MonoDevelop to avoid problems with wild MS updates, as shown in the past.

minimum 2 msbuild fallback versions:

one for dotnetcore (building 2.1,3.1,5 ff)

one for .net framework . a mono framework version at least is in https://github.com/dotnet/msbuild/blob/vs16.9 if FullFrameworkTFM or MonoBuild is set.

**Comment by hwthomas** [comment]
Up until ``dotdevelop_oe_8.1.5``, only Mono-MSBuild was required for the build.  All versions since then have needed ``dotnet`` installing as well in order to build.  In the longer term, is the aim to be able to build with *just* dotnet? And if so, will Mono-MSBuild (and Mono as the runtime, perhaps) become redundant?

**Comment by lytico** [comment]
i don't think it becomes redundant. it's .net48, and this has no end of lifetime (currently)
so people (including me for some conservative customers, or why ever they insistence on .netframework) has to deliver .net48-versions.

**Comment by lytico** [comment]
omnisharp-vscode manages it somehow to work with .netframework: 
https://github.com/OmniSharp/omnisharp-vscode/wiki/Desktop-.NET-Framework
as far as i know, it's distributed with a omnisharp local mono build.

take a look - if you have vscode with c#-extensions installed - in a folder like this:

~/.vscode/extensions/ms-dotnettools.csharp-1.23.8

and you'll find a small mono distri inside.

maybe we can reuse a pattern like this.
